packages/cardano-services/src/Http/util.ts

```diff
@@ -0,0 +1,19 @@
+import { ProviderError, ProviderFailure } from '@cardano-sdk/core';
+
+export const ORIGIN = 'Origin';
+export const CONTENT_TYPE = 'Content-Type';
+export const APPLICATION_JSON = 'application/json';
+
+export const getListen = (url: URL) => ({ host: url.hostname, port: url ? Number.parseInt(url.port) : undefined });
+
+type StaticOrigin = boolean | string | RegExp | (boolean | string | RegExp)[];
+
+export const corsOptions = (allowedOrigins: string[]) => ({
+  origin(requestOrigin: string | undefined, callback: (err: Error | null, options?: StaticOrigin) => void) {
+    if (requestOrigin && allowedOrigins.includes(requestOrigin)) {
```

Comment:
If cors middleware is enabled (`ALLOWED_ORIGINS` is provided as runtime configuration ) then the provider server will expect all incoming requests to have an origin header, otherwise, they will be rejected as described in the PR notes

Reply:
Sorry,  I misread it